### ISSUE TITLE
Error popup when invoking edit

### ISSUE THREAD

**Issue by JTurco32** [jrincayc/ucblogo-code#113]
![2021-08-13-160600_1920x1080_scrot](https://user-images.githubusercontent.com/87979348/129413164-9c3c506a-b89a-4f4a-a79d-2749767944b7.png)
hello,

i am learning logo, and came by this when invoking edit in the interpreter:

````
ASSERT INFO:
./src/gtk/textctrl.cpp(1917): assert ""start >= 0 && end <= l"" failed in SetStyle(): invalid range in wxTextCtrl::SetStyle

BACKTRACE:
[1] wxTextCtrl::SetStyle(long, long, wxTextAttr const&)
[2] wxEvtHandler::ProcessEventIfMatchesId(wxEventTableEntryBase const&, wxEvtHandler*, wxEvent&)
[3] wxEventHashTable::HandleEvent(wxEvent&, wxEvtHandler*)
[4] wxEvtHandler::TryHereOnly(wxEvent&)
[5] wxEvtHandler::ProcessEventLocally(wxEvent&)
[6] wxEvtHandler::ProcessEvent(wxEvent&)
[7] wxEvtHandler::SafelyProcessEvent(wxEvent&)
[8] wxTextEntryBase::SendTextUpdatedEvent(wxWindow*)
[9] wxTextEntry::DoSetValue(wxString const&, int)
[10] wxTextAreaBase::DoLoadFile(wxString const&, int)
[11] wxEntry(int&, wchar_t**)
[12] __libc_start_main
````
attached is a screenshot of the popup.

**Comment by dmalec** [comment]
Hello,

Welcome! I hope you enjoy learning Logo as much as I do :) It looks like you are running on a Raspberry Pi? 

Looking at this error, it's possible it is the same as issue #33 (which is fixed in the master branch; but, is not fixed in the versioned releases). Are you running a versioned release or did you compile from the master branch?

**Comment by JTurco32** [comment]
hey!

Im using the versioned release. I couldn't get the master branch to compile
on my raspberry pi (64 bit).

**Comment by dmalec** [comment]
Gotcha on the versioned release - that would still have issue #33 present in it.

If you're interested in giving compiling from the master branch another try, I did write up a blog post when I got my Pi 400 last year:
https://danmalec.com/2020/11/30/building-ucblogo-berkeley-logo-on-the-raspberry-pi-400/

To make sure it still works, I just took a fresh image of the latest Raspberry Pi OS and walked through it - all looks to still be accurate.

**Comment by JTurco32** [comment]
@dmalec thanks ill check it out. Sorry to derail the topic, but do you know if i could build this on mac? Im more versed with linux stuff but not so much with Mac OS. I Think i need xcode to build C packages? If you don't know that's ok. The current version for mac (Brians 6.2) has the same issue with edit, but instead of a popup you can bypass, it just shuts down logo.

**Comment by dmalec** [comment]
Agreed, the way the bug manifests on Mac is more extreme (that came in as issue #89)

ucblogo will build from the terminal CLI on Mac OS; though I don't have as clear a memory of how I set up my build environment... I did already have Xcode installed so I'm unsure how much Xcode had already provided needed tooling.  One other wrinkle, on Mac OS, building _wxWidgets_ from source is part of the workflow.

I did run the following prior to my first time building to get the build tooling setup (and had already installed `brew` itself by then):
```
brew install autoconf automake libtool autoconf-archive
```

The steps for building on Mac OS (including building _wxWidgets_) are captured in the wiki here: https://github.com/jrincayc/ucblogo-code/wiki/package_creation

One caveat, on my setup, I also pass `--disable-docs` to `./configure` as I have yet to get a working setup going on my Mac for the doc building toolchain.
